cpp/modules/math.gl/core/src/core.h

```diff
@@ -123,12 +125,16 @@ auto operator!=(const Vector2<coord> &v1, const Vector2<coord> &v2) -> bool {
 ///////////////////////////////////////////////////////////
 //  Vector3
 
+// TODO(ilija@unfolded.ai): When using std140 memory layout in GLSL vec3 takes in 16bytes
+// We should either use vec3 in our shaders instead, or create a GPUVector3 class that we'd use instead
 template <typename coord>
-class Vector3 {
+class alignas(16) Vector3 {
  public:
   Vector3() : x(0), y(0), z(0) {}
   Vector3(coord x1, coord y1, coord z1) : x(x1), y(y1), z(z1) {}
   Vector3(Vector2<coord> xy, coord z1) : x(xy.x), y(xy.y), z(z1) {}
+  template <typename othercoord>
+  Vector3(const Vector3<othercoord> &other);
```

Comment:
This will enable silent conversion between types, not necessarily a good idea. Consider adding `explicit` keyword.